runtime/shared_common/shrinit.cpp

```diff
@@ -3397,7 +3397,7 @@ j9shr_init(J9JavaVM *vm, UDATA loadFlags, UDATA* nonfatal)
 
 	cmBytes = SH_CacheMap::getRequiredConstrBytes(false);
 	nameBytes = (strlen(modifiedCacheNamePtr)+1) * sizeof(char);
-	modContextBytes = modContext ? ((strlen(modContext) * sizeof(char)) + sizeof(J9UTF8)) : 0;
+	modContextBytes = modContext ? (((strlen(modContext) + 1) * sizeof(char)) + sizeof(J9UTF8)) : 0;
```

Comment:
Why the `+ 1`? The characters in `J9UTF8` are not expected to be nul-terminated.

Reply:
For whatever reason, the code below is using `strcpy` to copy the name, so it is terminating it. This was hidden before by the fact that `sizeof(J9UTF8)` included two extra bytes.

Reply:
Ideally we would fix all the places that (needlessly) allocate room for terminators, but this isn't the time for that.

Reply:
Those terminators are a cheap way to make the strings easy to work with in the debugger.  The null byte is low overhead for some easier debug-ability and better C string interop.

Reply:
That just inverts the issue into one where terminators are added where they're currently missing.